### PR TITLE
 Represent Objective-C block pointers in methods as Kotlin functions

### PR DESCRIPTION
**Pull request by SvyatoslavScherbina**
Also 

* Enable method signature checks when overriding Objective-C method;
* Make linkage issues more understandable in Objective-C interop;
* Do some refactoring.

### CODE REVIEW

File: Interop/Runtime/src/native/kotlin/kotlinx/cinterop/ObjectiveCImpl.kt [hunk 1]
```diff
@@ -124,6 +138,19 @@ annotation class InteropStubs()
 @Retention(AnnotationRetention.SOURCE)
 private annotation class ObjCMethodImp(val selector: String, val encoding: String)
 
+@konan.internal.ExportForCppRuntime("Kotlin_Interop_getObjCClass")
+private fun getObjCClassByName(name: NativePtr): NativePtr {
+    val result = objc_lookUpClass(name)
+    if (result == nativeNullPtr) {
+        val className = interpretCPointer<ByteVar>(name)!!.toKString()
+        val message = "Objective-C class '$className' not found.\n" +
```
Comment (olonho) [inline Interop/Runtime/src/native/kotlin/kotlinx/cinterop/ObjectiveCImpl.kt]: Why not multiline literal?

File: Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/TypeUtils.kt [hunk 2]
```diff
@@ -51,6 +51,7 @@ fun Type.getStringRepresentation(): String = when (this) {
         is ObjCClassPointer -> "Class$protocolQualifier"
         is ObjCObjectPointer -> "${def.name}$protocolQualifier*"
         is ObjCInstanceType -> TODO(this.toString()) // Must have already been handled.
+        is ObjCBlockPointer -> "id"
```
Comment (olonho) [inline Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/TypeUtils.kt]: Is name conflict possible here?

File: runtime/src/main/cpp/ObjCInteropUtils.mm [hunk 3]
```diff
@@ -103,6 +103,50 @@ id MissingInitImp(id self, SEL _cmd) {
   return nullptr;
 }
 
+// TODO: optimize
```
Comment (olonho) [inline runtime/src/main/cpp/ObjCInteropUtils.mm]: Optimize how?

File: Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/TypeUtils.kt (hunk 2 above)
Comment (SvyatoslavScherbina) [reply Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/TypeUtils.kt]: The names returned from this function are not to be used to declare types, but just to mention them. And `id` is the builtin Objective-C type, with meaning similar to `void*`.

**Pull request merged**